### DISCUSSION TITLE
LaTeX Error: File `sjtuthesis.cls' not found. Experienced on latex.sjtu

### DISCUSSION THREAD

**Discussion by sjtugzx** [sjtug/SJTUThesis#1027] [提问]
I have come across "LaTeX Error: File `sjtuthesis.cls' not found. " via using the latex template on latex.sjtu . My settings followed the suggestions in readme file with xelatex compiler. 
<img width="856" alt="截屏2024-03-26 09 34 30" src="https://github.com/sjtug/SJTUThesis/assets/20066885/60ea9c04-76ab-4656-8db3-b48b4c4a976d">
<img width="824" alt="截屏2024-03-26 09 38 44" src="https://github.com/sjtug/SJTUThesis/assets/20066885/8675f18e-ffa8-4594-b1e7-afecb3aa87e5">

**Comment by LogCreative** [comment]
https://github.com/sjtug/SJTUThesis/discussions/980#discussioncomment-8909683

**Comment by zxhex** [comment]
在Visual Studio Code (VSCode) 中更改默认编译器为 `latexmk` 并添加 `-xelatex` 参数到默认命令行参数，你可以按照以下步骤操作：

1. 打开VSCode。

2. 打开命令面板，可以通过按下 `Ctrl+Shift+P`（在Mac上是 `Cmd+Shift+P`）。

3. 在命令面板中输入或选择 "Preferences: Open Settings (JSON)" 来打开用户设置的JSON文件。

4. 在打开的JSON文件中，你可以添加或修改以下设置来配置LaTeX工作流：

```json
{
  "latex-workshop.latex.recipes": [
    {
      "name": "xelatex",
      "tools": [
        "xelatex"
      ]
    }
  ],
  "latex-workshop.latex.tools": [
    {
      "name": "xelatex",
      "command": "latexmk",
      "args": [
        "-xelatex",
        "-synctex=1",
        "-interaction=nonstopmode",
        "-file-line-error",
        "-outdir=%OUTDIR%",
        "%DOC%"
      ]
    }
  ],
  "latex-workshop.latex.recipe.default": "xelatex"
}
```

5. 确保你已经安装了 `latexmk` 和 `xelatex`。如果没有，你需要先安装它们。通常，这些工具会包含在TeX发行版中，比如TeX Live或MiKTeX。

6. 保存设置文件。

7. 重启VSCode以使设置生效。

这样，你就成功地将 `latexmk` 作为默认编译器，并添加了 `-xelatex` 参数到命令行参数中。当你使用VSCode编译LaTeX文档时，它将使用这个配置。

实测有效
